k8s/claim/async_deprovision.go

```diff
@@ -28,6 +28,18 @@ func pollDeprovisionState(
 	pollNum := 0
 	pollState := mode.LastOperationStateInProgress
 	pollErrCount := 0
+
+	cfg, err := getConfig()
```

Comment:
iirc some config is fetched a few stack frames above this function. why not merge this new config struct with that one and pass the max async duration down to here?